cluster_tools/cluster_tools/schedulers/slurm.py

```diff
@@ -466,30 +478,33 @@ def _investigate_time_limit(
         return (reason, RemoteTimeLimitException)
 
     def _investigate_memory_consumption(
-        self, properties: dict[str, str]
+        self, stdout: str
     ) -> tuple[str, type[RemoteOutOfMemoryException]] | None:
-        if not properties.get("Memory Efficiency", None):
-            return None
-
-        # Extract the "25019.18% of 1.00 GB" part of the line
-        efficiency_note = properties["Memory Efficiency"]
-        PERCENTAGE_REGEX = r"([0-9]+(\.[0-9]+)?)%"
-
-        # Extract the percentage to see whether it exceeds 100%.
-        match = re.search(PERCENTAGE_REGEX, efficiency_note)
-        percentage = None
-        if match is None:
-            return None
-
-        try:
-            percentage = float(match.group(1))
-        except ValueError:
-            return None
-
-        if percentage < 100:
+        stdout_lines = stdout.splitlines()
```

Comment:
could you add a comment that mentions the expected format for the stdout input? this makes it easier to follow the parsing logic.